src/Compilers/Core/Portable/DiagnosticAnalyzer/ShadowCopyAnalyzerAssemblyLoader.cs

```diff
@@ -129,64 +138,65 @@ private void DeleteLeftoverDirectories()
             }
         }
 
-        protected override string PreparePathToLoad(string originalAnalyzerPath)
-        {
-            var mvid = AssemblyUtilities.ReadMvid(originalAnalyzerPath);
-
-            return PrepareLoad(_mvidPathMap, mvid, copyAnalyzerContents);
+        public bool IsAnalyzerPathHandled(string analyzerFilePath) => true;
 
-            string copyAnalyzerContents()
-            {
-                var analyzerFileName = Path.GetFileName(originalAnalyzerPath);
-                var shadowDirectory = Path.Combine(_shadowCopyDirectoryAndMutex.Value.directory, mvid.ToString());
-                var shadowAnalyzerPath = Path.Combine(shadowDirectory, analyzerFileName);
-                CopyFile(originalAnalyzerPath, shadowAnalyzerPath);
-
-                return shadowAnalyzerPath;
-            }
+        public string GetRealAnalyzerPath(string analyzerFilePath)
+        {
+            var analyzerShadowDir = GetAnalyzerShadowDirectory(analyzerFilePath);
+            var analyzerShadowPath = Path.Combine(analyzerShadowDir, Path.GetFileName(analyzerFilePath));
+            _ = _shadowToOriginalPathMap[analyzerShadowPath] = analyzerFilePath;
```

Comment:
/seinfeld: what's the deal with the discard here for assignment into a dict?

Reply:
At some point that must have been a `TryAdd`. I use discards there to be explicit that I understand it returns a `bool` but it does not matter in that code path. Basically, don't care if I won or not, just want "first write wins" behavior.